src/pudl/analysis/ferc1_eia.py

```diff
@@ -0,0 +1,1529 @@
+        # the plant records so we have all possible FERC plant records, even
+        # the unmapped ones
+        .merge(
+            plants_ferc1_df,
+            how="outer",
+            on=["record_id_ferc1", "report_year", "plant_id_pudl", "utility_id_pudl"],
+            suffixes=("_eia", "_ferc1"),
+            validate="1:1",
+            indicator=True,
+        ).assign(
+            report_date=lambda x: pd.to_datetime(
+                x.report_year, format="%Y", errors="coerce"
+            ),
+        )
+    )
+
+    no_ferc = connects_ferc1_eia[
+        (connects_ferc1_eia._merge == "left_only")
+        & (connects_ferc1_eia.record_id_eia.notnull())
+        & ~(connects_ferc1_eia.record_id_ferc1.str.contains("_hydro_", na=False))
+        & ~(connects_ferc1_eia.record_id_ferc1.str.contains("_gnrt_plant_", na=False))
+    ]
+    connects_ferc1_eia = connects_ferc1_eia.drop(columns=["_merge"])
+    if not no_ferc.empty:
+        message = (
+            "Help. \nI'm trapped in this computer and I can't get out.\n"
+            ".... jk there shouldn't be any matches between FERC and EIA\n"
+            "that have EIA matches but aren't in the FERC plant table, but we\n"
+            f"got {len(no_ferc)}. Check the training data and "
+            "prettyify_best_matches()"
+        )
+        if debug:
+            warnings.warn(message)
+            return no_ferc
+        else:
+            logger.info(
+                "jsuk there are some FERC-EIA matches that aren't in the steam \
+                table but this is because they are linked to retired EIA generators."
+            )
+            # raise AssertionError(message)
+            warnings.warn(message)
+
+    _log_match_coverage(connects_ferc1_eia)
+    for match_type in ["all", "overrides"]:
+        check_match_consistency(
+            connects_ferc1_eia,
+            train_df,
+            match_type=match_type,
+        )
+
+    return connects_ferc1_eia
+
+
+def _log_match_coverage(connects_ferc1_eia):
+    eia_years = DataSource.from_id("eia860").working_partitions["years"]
+    # get the matches from just the EIA working years
+    m_eia_years = connects_ferc1_eia[
+        (connects_ferc1_eia.report_date.dt.year.isin(eia_years))
+        & (connects_ferc1_eia.record_id_eia.notnull())
+    ]
+    # get all records from just the EIA working years
+    r_eia_years = connects_ferc1_eia[
+        connects_ferc1_eia.report_date.dt.year.isin(eia_years)
+    ]
+
+    fuel_type_coverage = len(
+        m_eia_years[m_eia_years.energy_source_code_1.notnull()]
+    ) / len(m_eia_years)
+    tech_type_coverage = len(
+        m_eia_years[m_eia_years.technology_description.notnull()]
+    ) / len(m_eia_years)
+
+    def _get_subtable(table_name):
+        return r_eia_years[r_eia_years.record_id_ferc1.str.contains(f"{table_name}")]
+
+    def _get_match_pct(df):
+        return round(len(df[df["record_id_eia"].notna()]) / len(df))
```

Comment:
This `round` is rounding to either 1 or 0 since the number is rounding is going to be between 0 and 1. I think you can get rid of this round entirely since the logged coverage below is rounding to .01 in the f string.